packages/workflow/src/workflow.ts

```diff
@@ -334,73 +330,75 @@ function signalWorkflowNextHandler({ seq, signalName, args, target }: SignalWork
   });
 }
 
-function activityInfo(activity: string | [string, string] | ActivityFunction<any, any>): ActivityInfo {
-  if (typeof activity === 'string') {
-    return { name: activity, type: activity };
-  }
-  if (activity instanceof Array) {
-    return { name: activity[1], type: JSON.stringify(activity) };
-  } else {
-    return activity as InternalActivityFunction<any, any>;
-  }
-}
-
 export class ContextImpl {
   /**
    * @protected
    */
   constructor() {
     // Does nothing just marks this as protected for documentation
   }
+
   /**
-   * Configure an activity function with given {@link ActivityOptions}
-   * Activities use the worker options's {@link WorkerOptions.activityDefaults | activityDefaults} unless configured otherwise.
+   * Configure Activity functions with given {@link ActivityOptions}.
    *
-   * @typeparam P type of parameters of activity function, e.g `[string, string]` for `(a: string, b: string) => Promise<number>`
-   * @typeparam R return type of activity function, e.g `number` for `(a: string, b: string) => Promise<number>`
+   * This method may be called multiple times to setup Activities with different options.
    *
-   * @param activity either an activity name if triggering an activity in another language, a tuple of [module, name] for untyped activities (e.g. ['@activities', 'greet']) or an imported activity function.
-   * @param options partial {@link ActivityOptions} object, any attributes provided here override the provided activity's options
+   * @return a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)
+   *         for which each attribute is a callable Activity function
+   *
+   * @typeparam A An {@link ActivityInterface} - mapping of name to function
    *
    * @example
    * ```ts
-   * import { Context } from '@temporalio/workflow';
-   * import { httpGet } from '@activities';
+   * import { Context, ActivityInterface } from '@temporalio/workflow';
+   * import * as activities from '../activities';
    *
-   * const httpGetWithCustomTimeout = Context.configure(httpGet, {
+   * // Setup Activities from module exports
+   * const { httpGet, otherActivity } = Context.configureActivities<typeof activities>({
    *   type: 'remote',
-   *   scheduleToCloseTimeout: '30 minutes',
+   *   startToCloseTimeout: '30 minutes',
    * });
```

Comment:
Would be nice to show setting different options for activities from the same import like in my question you answered in the proposal since that wasn't immediately obvious to some of us.

Reply:
I'll add that in another PR